packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -23,7 +23,118 @@ const restrictedNodeFields = [
 
 exports.setFieldsOnGraphQLNodeType = require(`./extend-node-type`).extendNodeType
 
-exports.onPreBootstrap = validateOptions
+const validateContentfulAccess = async pluginOptions => {
+  if (process.env.NODE_ENV === `test`) return undefined
+
+  await fetch(
+    `https://${pluginOptions.host}/content/v1/spaces/${pluginOptions.spaceId}`,
```

Comment:
So this adds an extra round-trip? For builds only. It's not entirely clear to me what the advantage is of this step. It appears to merely test whether the connect can be created or not, the results are dropped otherwise. Wouldn't the user be notified of this when making the actual connection?

Reply:
Most errors I've seen are due to options being missing completely. The next step from there is that the schema options are present but the access to the space is missing. This check would allow us to fail earlier

Reply:
It'll also give the user a more specific error. Rather than something cryptic like "Field `allContentfulNode` doesn't exist", they'll now be warned early that their access token and space ID don't match / are incorrect!

Reply:
@mxstbr this is one of the _most_ common errors I see, and failing fast and short-circuiting with a far more helpful (and fixable) error here is bomb. Nicely done!

Reply:
Mismatch of HOST + Access token is a super common support request - if we surface this more clearly in the UI that would be awesome!

Reply:
This is 👍 We check this during source-nodes so testing it here is better.